pyfftw/interfaces/_utils.py

```diff
@@ -97,24 +101,25 @@ def _Xfftn(a, s, axes, overwrite_input, planner_effort,
         # take the transform of the input array! (in general, we have
         # to assume that the input array will be destroyed during 
         # planning).
-        a_copy = a.copy()
+        if overwrite_input:    # Note: reload_after_transform is not True...
+            args = (a, s, axes) + args
+            a_copy = a.copy()  # ... hence this is not an additional copy
+        else:
+            args = (numpy.empty_like(a), s, axes) + args
```

Comment:
`A.strides == B.strides` returns `False`. The FFTW object that is created for A (or B), cannot then be used to process the other.

It's probably possible to create an array with the same strides using `numpy.lib.stride_tricks.as_strided`, but it would need a bit of thinking about to work robustly.